lib/editor/insert-or-remove-checkboxes.js

```diff
@@ -0,0 +1,117 @@
+import { EditorState, Modifier, SelectionState } from 'draft-js';
+import { get } from 'lodash';
+
+import { taskRegex } from '../note-detail/toggle-task/constants';
+
+/**
+ * Insert/remove checkbox on current line(s)
+ *
+ * @param {EditorState} editorState - The current editor state.
+ * @returns {EditorState} The new editor state.
+ */
+export default function insertOrRemoveCheckboxes(editorState) {
+  const contentState = editorState.getCurrentContent();
+  const selection = editorState.getSelection();
+  let blockKey = selection.getStartKey();
+  const endBlockKey = selection.getEndKey();
+  let newContentState = contentState;
+
+  while (blockKey) {
+    newContentState = insertOrRemoveForBlock(blockKey, newContentState);
+
+    if (blockKey === endBlockKey) {
+      break;
+    }
+
+    blockKey = contentState.getKeyAfter(blockKey);
+  }
+
+  const newEditorState = EditorState.push(
+    editorState,
+    newContentState,
+    'insert-characters'
+  );
+  const adjustedSelection = adjustSelectionState(
+    selection,
+    contentState,
+    newContentState
+  );
+
+  return EditorState.forceSelection(newEditorState, adjustedSelection);
+}
+
+/**
+ * Insert/remove checkbox in a given ContentBlock
+ *
+ * @param {string} blockKey - The key of the target ContentBlock.
+ * @param {ContentState} contentState - The current content state.
+ *
+ * @returns {ContentState} The new content state.
+ */
+export function insertOrRemoveForBlock(blockKey, contentState) {
+  const text = contentState.getBlockForKey(blockKey).getText();
+  taskRegex.lastIndex = 0; // Reset the search
+  const matches = taskRegex.exec(text);
+  let newContentState;
+
+  if (matches) {
+    // Remove task prefix
+    const leadingWhitespace = get(matches, '1', '');
+    const taskPrefix = matches[2];
+    const spacesToTrim = matches[3].match(/^\s*/)[0].length;
```

Comment:
Are we safe from out of index exceptions here?

Reply:
Should be ok, and unit tests will fail spectacularly if the regex is changed in an unexpected way. I tweaked the regex a bit to ensure that the first capturing group is always a string though (https://github.com/Automattic/simplenote-electron/pull/1166/commits/0a738e6a49d1ae8d22ce0c549003d335e3e7fb63).